### ISSUE TITLE
no-import-private: substring "/private" matching causes false positives and breaks on Windows

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#544]
### Describe the bug

`no-import-private` decides "is this a private-dir import" with plain substring/split operations on `"/private"` and splits paths with a hardcoded token, which produces several false positives and breaks the rule entirely on Windows:

**1. Directory names that merely start with `private` are treated as private dirs.** `importPath.includes("/private")` and `absoluteImportPath.split("/private")[0]` match `/private-utils`, `/private_helpers`, etc.

```ts
// src/moduleA/from-private-utils.ts
import { helper } from "../shared/private-utils/helper"; // not a `private` dir, but reported
```

**2. Importers living inside a `private` subtree cannot use their own `private` dir.** `split("/private")[0]` cuts at the **first** occurrence, so for a file `src/lib/private/helper/x.ts` importing `./private/y`, the comparison prefix becomes `src/lib` instead of `src/lib/private/helper`, and this documented-as-correct same-level import is reported.

```ts
// src/lib/private/helper/x.ts
import { y } from "./private/y"; // same-level private import (docs: Correct), but reported
```

**3. Windows: every `/private` import is reported, including same-level ones.** On Windows `path.resolve` produces `\`-separated paths, `split("/private")` never matches, so `importDirBeforePrivate` keeps the full module path and the segment comparison always fails.

### Expected Behavior

- 1, 2: no report — `private-utils` is not a `private` directory, and `./private/y` from within `src/lib/private/helper/` is exactly the pattern the docs show as correct.
- 3: same behavior as on POSIX.

### Current Behavior

ESLint on macOS (cases 1 and 2; `moduleA/index.ts:2` is a genuine cross-module control):

```
repro/no-import-private/src/lib/private/helper/x.ts    3:1  error  Cannot import modules from private dir at different levels of the hierarchy  awscdk/no-import-private
repro/no-import-private/src/moduleA/from-private-utils.ts  2:1  error  Cannot import modules from private dir at different levels of the hierarchy  awscdk/no-import-private
repro/no-import-private/src/moduleA/index.ts           2:1  error  Cannot import modules from private dir at different levels of the hierarchy  awscdk/no-import-private
```

(Controls behaved correctly: `./private/own` and a non-private import in `moduleA/index.ts` were not reported.) Oxlint reports the identical set.

Case 3, verified by running the actual built rule with `path` rebound to `path.win32` (simulation on macOS, not a real Windows machine):

```
same-level "./private/c" from C:\proj\src\sampleB\a.js (expected: no report)
  => ["invalidImportPath"]
cross-level "../sampleB/private/c" from C:\proj\src\sampleA\a.js (expected: report)
  => ["invalidImportPath"]
```

### Plugin

eslint-plugin-awscdk, oxlint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.0, oxlint-plugin-awscdk 1.0.0

### Language Version

TypeScript 6.0.3 (eslint-plugin-awscdk) / TypeScript 7 preview via oxlint-tsgolint (oxlint-plugin-awscdk)

### Additional Context

- Verified on `main` (35b889a); the rule implementation is identical in both packages (`packages/*/src/rules/no-import-private.ts`).
- Root cause: matching should be segment-based, e.g. split the resolved path with `path.sep` and look for a segment exactly equal to `private` (the **last** such segment relative to the import target), instead of `includes("/private")` / `split("/private")`.
- Related: bare specifiers containing `/private` (e.g. `import x from "@myorg/private-utils"`) are also resolved as file paths relative to the importing file and reported; a segment-based check plus skipping non-relative specifiers would fix that too (not runtime-verified, same code path as case 1).
